lib/uplink/uplink.go

```diff
@@ -92,7 +92,7 @@ func (cfg *Config) setDefaults(ctx context.Context) error {
 		cfg.Volatile.MaxMemory = 0
 	}
 	if cfg.Volatile.Log == nil {
-		cfg.Volatile.Log = zap.NewNop()
+		cfg.Volatile.Log = zap.L()
```

Comment:
The reason that this was not the default was because of C bindings that shouldn't use stdout or stderr.

Reply:
I'm not sure I follow but we should probably solve that is a way that doesn't shut off logging for the rest of the system

Reply:
When something needs logging, then it would need to explicitly enable it by passing in the appropriate logger. The default should probably still be nop.

Reply:
I think logging should be on by default, if something wants to shut off logging, it can pass in a no op logger

Reply:
I disagree with that, our things are intended to be used as libraries not as a program. Every program has their own logging and concerns to worry about, the log of our library code shouldn't be their concern.

Imagine Go runtime or GRPC or HTTP2 all logging by default, it doesn't make sense, at least for me.

Reply:
Should "libraries" log at all ?

Reply:
I think they should be quiet by default. Vital information is communicated through errors.

Reply:
I think uplink is a service not a library and to have it not show errors by default will be confusing and lead to issues, like we experienced.

Reply:
`uplink`, yes, but not `libuplink`...

Reply:
I think the default behavior of no logger is counter intuitive for a "library" like libuplink. It caused us to not log anything and took time to track down what happened. There are well regarded open source projects that log by default and I think we should keep it like this until there is good practical reasons for not doing so.

Reply:
I'd have to disagree. If we're calling the thing "libuplink", then people will treat it as a library and have library expectations for it. Libraries should never log without being explicitly configured to do so.

But I'm all for making it easy to see how that should be done in our docs, sample code, etc.

Reply:
1. You may not always have stdout/stderr available
2. C handles buffering differently than Go
3. Our code is concurrent and uses different logging framework which will interrupt stdout output
4. as far as I know, S3, GCS packages don't log

Reply:
@egonelbre Can you provide the correct solution that resolves the problem of not logging at all in uplink and gateway?

Reply:
>  Libraries should never log without being explicitly configured to do so.
Agreed 100% -- I guess my argument is uplink feels like a service not a library (we aren't calling it libuplink it's just int he lib folder, maybe it shouldn't be) but I lack any good reason for that other than having a config file for a library seems unusual.

Reply:
Specify:

```
config.Volatile.Log = zap.L()
libuplink.NewUplink(ctx, config)

or 

config.Volatile.Log = parentLogger
libuplink.NewUplink(ctx, config)
```

Reply:
the issue is the default when there is no parent

Reply:
Talked with @thepaul offline. I'll make another PR adding loggers to calls to NewUplink and swap the default back